server/src/config.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) ZeroC, Inc.
+
+use crate::Backend;
+use tower_lsp::lsp_types::{ConfigurationItem, DidChangeConfigurationParams, Url};
+
+#[derive(Default, Debug)]
+pub struct SliceConfig {
+    pub reference_urls: Option<Vec<Url>>,
+}
+
+impl SliceConfig {
+    // Attempt to create a SliceConfig from the backend with the provided uri root.
+    pub async fn try_from_backend(
+        backend: &Backend,
+        root_uri: &Url,
+    ) -> tower_lsp::jsonrpc::Result<SliceConfig> {
+        let reference_directories = Self::fetch_reference_directories(backend).await?;
+        Self::create_config_from_directories(reference_directories, root_uri)
+    }
+
+    // Creates a SliceConfig from the provided configuration parameters and uri root.
+    pub fn from_configuration_parameters(
+        params: DidChangeConfigurationParams,
+        root_uri: &Url,
+    ) -> Self {
+        let reference_directories = Self::parse_reference_directories(&params);
+        Self::create_config_from_directories(reference_directories, root_uri).unwrap_or_default()
+    }
+
+    // Fetch reference directories from the backend.
+    async fn fetch_reference_directories(
+        backend: &Backend,
```

Comment:
I think it would be clearer for this function to take the client, which is really what it's looking for to get the configuration.